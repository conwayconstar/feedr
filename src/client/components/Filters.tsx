import React from 'react';

export type FiltersProps = {
  search: string | '';
  setSearch: (search: string) => void
};

const Filters: React.FC<FiltersProps> = ({ search, setSearch }) => (
  <div className="filters">
    <input
      className="form-control"
      placeholder="Name"
      value={search}
      onChange={({ target: { value } }) => setSearch(value)}
    />
  </div>
);

export default Filters;
