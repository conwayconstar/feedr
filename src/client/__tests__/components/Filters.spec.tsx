import { shallow } from 'enzyme';
import React from 'react';
import Filters, { FiltersProps } from '../../components/Filters';

describe('Filters component', () => {
  const setSearch = jest.fn((search) => search);

  const props: FiltersProps = {
    search: 'Quinoa',
    setSearch,
  };
  const wrapper = shallow(<Filters {...props} />);

  it('Filters should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('setSearch should be called', () => {
    const value = 'Cuban Beef';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'Cuban Beef' } });
    expect(setSearch.mock.results[0].value).toEqual(value);
  });
});
