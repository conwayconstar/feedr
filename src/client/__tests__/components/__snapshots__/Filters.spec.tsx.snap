// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component Filters should match snapshot 1`] = `
<div
  className="filters"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Name"
    value="Quinoa"
  />
</div>
`;
